### DISCUSSION TITLE
Config File

### DISCUSSION THREAD

**Discussion by Aece96** [vitrivr/cineast#363] [Q&A] [answered]
Dear community

I am trying to set up the vitrivr software stack (vitrivr-ng, cineast and cottontaildb) and do some initial uploads and extractions.
However I seem to miss a config file and can t find anything related in the given wikis or the discussions available. The error message that local host throws me looks like this:
![Config missing](https://user-images.githubusercontent.com/45922740/217372062-af20cf66-bc25-479c-9da9-caa9ca39ce7c.png)

Has anybody an idea where I have gone wrong, what I am missing?
Thank you very much for your help in advance :)
Best Cyrill

**Comment by sauterl** [comment, marked as answer]
Hi there

We do follow a zero config approach, so it _should_ ™️  work out of the box.
However, this particular error might also hint (although slightly less obvious) to the fact that your `config.json` is not in valid JSON format.

It might be worthwhile to a) present your project's files (i.e. which files are where, what is your setup) and b) share your `config.json` in case you have one.

**Reply by Aece96** [reply]
Hi again

Thank you very much for the fast answer. That is kind of my problem - I am having trouble to evaluate whether I miss a config file and if so how I can establish one. 

As far as files go - I have set up the system entirely new so the files are the default ones provided when gradlew is ran. I am working on a Windows OS. 

Basically when I set up cottontail db I get the following statement:
![grafik](https://user-images.githubusercontent.com/45922740/218491100-f00f72f2-cf86-473c-961c-e42189c5a8cc.png)
After that there are a lot of Log set up messages I do not quite understand yet.
![grafik](https://user-images.githubusercontent.com/45922740/218491475-c9e90421-d8a1-4f56-b3e8-43e8a2b76226.png)
However so far the Cottontail database seems to be working. I am not sure if I already have any data in it yet though. 
The config file that gets created is emtpy:
![grafik](https://user-images.githubusercontent.com/45922740/218497295-078011c5-3056-4e6b-b86b-084f2d8703c8.png)


The vitrivr ng frontend is up and running like in the presentation video. The only problem seems to be the connection with the retrieval engine cineast to connect the front end with the backend.

The setup for cineast was done in the following way:
1. cloning github
2. build external files via gradlew (command line in cineast and then gradlew getExternalFiles)
3. start the shadowAPI Jar (gradlew cineast-api:shadowJar)
4. starting cineast using the built shadow Jar (java -jar *path*/cineast-api-3.12.4-all.jar *either cineast.json or config.json*)

Sadly the initially mentioned failed connection with cineast was not resolved. This would be the command line output.
![grafik](https://user-images.githubusercontent.com/45922740/218495959-a99ed1a6-0310-4333-a3e6-ad0532658db3.png)


I have tried various times to adapt files which are in the root directory that were named config.json or cineast.json or anything in that regard. So from the looks of it I am missing a  config file that should be created automatically or I am missing some vital set up points and cant figure out where I have gone wrong.

Thank you very much for your help, I hope those screenshots are providing insight into my problem.

**Reply by sauterl** [reply]
Thanks for the more detailed information provided.

## CottontailDB

Your setup seems fine and ufortunately, there is this stack-trace with regards to logging that is annoying but is expected.
As far as 'data in cottontail' is concerned: I do think that, unless you specifically imported data somehow not mentioned, it will be empty.

## Cineast

This is a little tricky. To me it appears that you run the shadowJar from `<cineast-repo>/cineast-api/build/libs/`, which comes with a lot of issues. One being, that the [default config](https://github.com/vitrivr/cineast/blob/main/cineast.json) is not available (as it's expected to be in the current working directory). This is the main reason for your cineast failure.
However, in addition, once cineast is running, external files (which are expected to be in `<current-working-dir>/ressoruces/` wouldn't be properly resolved either.

Hence, I suggest that you'll start cineast from the repo's root and see how far you can get.

I kindly ask you to mark one of the answers as solution and open a new dicsussion, if you have further questions. All the best in setting the system up!

**Reply by silvanheller** [reply]
Additionally:

# Cottontail
- The errorlog is a known issue with Java 17, but does not affect functionality: https://github.com/vitrivr/cottontaildb/issues/119

# Cineast
- As mentioned by @sauterl, this is due to no config being available.

# Your original error
- This actually just means that no vitrivr-ng config file is available and the default config is used - this is not a problem!
